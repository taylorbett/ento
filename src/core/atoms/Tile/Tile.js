import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Tile = styled.div`
    display: ${props => props.shrinkWrap === true ? 'inline-block' : 'block'};
    border: 1px solid #e4e9ed;
    border-radius: 4px;
    background: #f8fafd;
    text-align: left;
    color: #1f1f24;
    position: relative;
    overflow: hidden;
    cursor: ${props => props.clickable ? 'pointer' : 'default'};
`;

export class TileComponent extends React.PureComponent {
    render() {
        console.log('rendering tile with', this.props);
        return (
            <Tile {...this.props}>
                {this.props.children}
            </Tile>
        );
    }
}

Tile.defaultProps = {
    children: null,
    shrinkWrap: false,
    clickable: false,
}

Tile.propTypes = {
    children: PropTypes.node,
    shrinkWrap: PropTypes.bool,
    clickable: PropTypes.bool,
};

export default TileComponent;