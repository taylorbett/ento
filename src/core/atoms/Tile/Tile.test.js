import React from 'react';
import Tile from './Tile';
import renderer from 'react-test-renderer';

describe('Tile', () => {
    it('renders correctly', () => {
        const clickable = false;
        const shrinkWrap = false;
        const tree = renderer
            .create(
                <Tile 
                    clickable={clickable}
                    shrinkWrap={shrinkWrap}
                />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});