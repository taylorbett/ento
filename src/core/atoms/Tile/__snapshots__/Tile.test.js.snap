// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile renders correctly 1`] = `
<div
  className="sc-bdVaJa cnZssB"
/>
`;
